// Brick/index.js
var  Resource = require('deployd/lib/resource')
	,util = require('util')
	,internalClient = require('deployd/lib/internal-client');

var dpd;
function Brick(name, options) {
    Resource.apply(this, arguments);
}
util.inherits(Brick, Resource);
module.exports = Brick;

Brick.prototype.clientGeneration = true;


Brick.prototype.handle = function (ctx, next) {
	try{
	    if(ctx.req && ctx.req.method == 'GET'){
	    	var brickType = ctx.query.type;
	    	var square = ctx.query.square;
				ctx.done(null, calculator(brickType, square));
	    } else return next();
	} catch(e){
		console.log(e);
	}
}



function Round2(val){
	val*=1000;
	val=Math.ceil(val);
	val=val/1000;
	return val;
}
function sootv(a){
	if (a==56.5)  return 0.0001920;
	if (a==114.3) return 0.0002375;
	if (a==170.8) return 0.0002568;
	if (a==228.6) return 0.0002812;
	if (a==285.1) return 0.0002341;
	if (a==42.16) return 0.0002058;
	if (a==86.1)  return 0.00026625;
	if (a==128.3) return 0.00025104;
	if (a==172.2) return 0.00029531;
	if (a==214.4) return 0.0003107;
	if (a==27.1) return 0.000237;
	if (a==55.2) return 0.00042187;
	if (a==82.3) return 0.0003052;
	if (a==110.4)return 0.00021875;
	if (a==137.5)return 0.00031885;
	return 1;
}
function calculator(brickType, square){
	calc_result	= Math.ceil(brickType * 1 * square *1);
	return {
		 bricks	: calc_result
		,rastv 	: Round2(calc_result * sootv(brickType * 1))
	}
}



process.server.on('listening', function() {
  dpd = internalClient.build(process.server);
});

/*
        <tr align=left>
            <td>Размер кирпича и кладки</td>
            <td><!-- value - расход на 1 м2 --><select id="f1" onchange="changeImage(value)" name="calc[f1]" style="width:220px">
            <option value="56.50" selected="selected">Кирпич одинарный, кладка в 0,5 кирпича, толщина стен - 120 мм</option>
            <option value="114.3">Кирпич одинарный, кладка в 1 кирпич, толщина стен - 250 мм</option>
            <option value="170.8">Кирпич одинарный, кладка в 1,5 кирпича, толщина стен - 380 мм</option>
            <option value="228.6">Кирпич одинарный, кладка в 2 кирпича, толщина стен - 510 мм</option>
            <option value="285.1">Кирпич одинарный, кладка в 2,5 кирпича, толщина стен - 640 мм</option>
            <option value="42.16">Кирпич полуторный, кладка в 0,5 кирпича, толщина стен - 120 мм</option>
            <option value="86.10">Кирпич полуторный, кладка в 1 кирпич, толщина стен - 250 мм</option>
            <option value="128.3">Кирпич полуторный, кладка в 1,5 кирпича, толщина стен - 380 мм</option>
            <option value="172.2">Кирпич полуторный, кладка в 2 кирпича, толщина стен - 510 мм</option>
            <option value="214.4">Кирпич полуторный, кладка в 2,5 кирпича, толщина стен - 640 мм</option>
            <option value="27.10">Камень двойной, кладка в 0,5 кирпича, толщина стен - 120 мм</option>
            <option value="55.20">Камень двойной, кладка в 1 кирпич, толщина стен - 250 мм</option>
            <option value="82.30">Камень двойной, кладка в 1,5 кирпича, толщина стен - 380 мм</option>
            <option value="110.4">Камень двойной, кладка в 2 кирпича, толщина стен - 510 мм</option>
            <option value="137.5">Камень двойной, кладка в 2,5 кирпича, толщина стен - 640 мм</option>
            </select> </td>
        </tr>*/
